service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -300,20 +394,19 @@ public Optional<EhrExtractStatus> updateEhrExtractStatusContinue(String conversa
     public void updateEhrExtractStatusAck(String conversationId, EhrReceivedAcknowledgement ack) {
 
         if (ack.getErrors() == null && !isEhrStatusWaitingForFinalAck(conversationId)) {
-            LOGGER.warn("Received unexpected acknowledgement of EHR Extract with conversation id=" + conversationId);
+            logger().warn("Received unexpected acknowledgement of EHR Extract with conversation id=" + conversationId);
             return;
         }
 
-        if (hasFinalAckBeenReceived(conversationId)) {
-            LOGGER.warn("Received an ACK message with a conversation_id=" + conversationId + " that is a duplicate");
+        if (hasAcknowledgementExceededEightDays(conversationId, ack.getReceived())
+            && hasEhrStatusReceivedAckWithUnexpectedConditionErrors(conversationId)) {
+
+            logger().warn("Received an ACK message with a conversation_id: {}, but it will be ignored", conversationId);
```

Comment:
I'd also be a bit clearer in this warning, to be clearer that the reason its being ignored is because we're already determined that the transfer has timed out because we didn't get an acknowledgement from the requesting practice in time.